### ISSUE TITLE
Bug: dot-notation false positive for property names containing hyphens

### ISSUE THREAD

**Issue by wagenet** [oxc-project/tsgolint#701]
### Description

`dot-notation` incorrectly flags bracket notation for property names that contain hyphens (or other characters that make them invalid JS identifiers), such as HTTP header names.

### Reproduction

```ts
const headers = response.headers;
const traceId = headers["X-Amzn-Trace-Id"];
```

**Diagnostic produced:**
```
typescript-eslint(dot-notation): ["X-Amzn-Trace-Id"] is better written in dot notation.
```

### Expected behavior

No diagnostic. `X-Amzn-Trace-Id` contains hyphens and is not a valid JavaScript identifier — `headers.X-Amzn-Trace-Id` would be parsed as `headers.X - Amzn - Trace - Id` (subtraction), not a property access.

### Root cause

Before suggesting dot notation, the rule should verify the property name matches the JS identifier pattern (`/^[a-zA-Z_$][a-zA-Z0-9_$]*$/` or equivalent). Any name containing hyphens, spaces, or other non-identifier characters must use bracket notation and should be exempt.

This is consistent with the behavior of the upstream [`@typescript-eslint/dot-notation`](https://typescript-eslint.io/rules/dot-notation/) rule and the base ESLint `dot-notation` rule.

**Comment by wagenet** [comment]
Additional reproduction: colons in property names trigger the same false positive.

```ts
obj["keyRiskIndicatorValue:originalStatus"]
```

**Diagnostic:**
```
typescript-eslint(dot-notation): ["keyRiskIndicatorValue:originalStatus"] is better written in dot notation.
```

`keyRiskIndicatorValue:originalStatus` contains a colon, making it an invalid identifier for the same reason. The fix should handle any character outside `[a-zA-Z0-9_$]` (plus the leading-digit restriction).
